index.js

```diff
@@ -8,9 +8,58 @@
 import './i18nInit';
 import { AppRegistry } from 'react-native';
 import { setJSExceptionHandler } from 'react-native-exception-handler';
+import messaging from '@react-native-firebase/messaging';
+import notifee, { EventType } from '@notifee/react-native';
 import App from './src/App';
 import { name as appName } from './app.json';
 import { errorHandler } from './src/errorHandler';
+import { messageHandler, setInitialNotificationData } from './src/sagas/pushNotification';
+
+/**
+ * This function is called when the app is in background or **quit** state
+ * and a push notification from **notifee** is received. It also handles
+ * notification events like dismiss and action press.
+ *
+ * @see https://notifee.app/react-native/reference/eventtype
+ *
+ * In this project the Notifee is used to interprete the data message received
+ * from firebase and show the notification to the user.
+ *
+ */
+const setNotifeeBackgroundListener = () => {
+  try {
+    notifee.onBackgroundEvent(async ({ type, detail }) => {
+      const { notification } = detail;
+
+      if (type === EventType.DISMISSED) {
+        await notifee.cancelNotification(notification.id);
+        return;
+      }
+
+      if (type === EventType.PRESS || type === EventType.ACTION_PRESS) {
+        setInitialNotificationData(notification);
+        await notifee.cancelNotification(notification.id);
+      }
+    });
+  } catch (error) {
```

Comment:
In that case it would fail silently, right? The app would work as usual and the user would never receive any push? If that's the case, I think we should proper handle this, as in the other possible push errors. (This comment is also for the method below)

Reply:
We could also display a non-fatal error modal, that is currently being developed in https://github.com/HathorNetwork/hathor-wallet-mobile/pull/231

Reply:
This is already merged, so we might just do it in this PR